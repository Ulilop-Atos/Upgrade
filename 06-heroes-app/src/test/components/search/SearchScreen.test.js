import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { SearchScreen } from "../../../components/search/SearchScreen";

const mockNavigate = jest.fn()

jest.mock('react-router-dom', 
  ()=>({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate,   
  })
);

describe('Tests for <SearchScreen />', () => {

  test('should display properly', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/search']}>
        <SearchScreen />
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.alert').text().trim()).toBe(`Search a hero`);
  });

  test('should use the query value as input and display a hero', () => {

    const hero = 'Spider Man';
    const wrapper = mount(
      <MemoryRouter initialEntries={[`/search?q=${hero}`]}>
        <SearchScreen />
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('input').prop('value')).toBe(hero);
    expect(wrapper.find('.card-title').text().trim()).toBe(hero);
  });

  test('should use the query value as input and display a not found message', () => {

    const hero = 'Crazy hero name';
    const wrapper = mount(
      <MemoryRouter initialEntries={[`/search?q=${hero}`]}>
        <SearchScreen />
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('input').prop('value')).toBe(hero);
    expect(wrapper.find('.alert-danger').text().trim()).toBe(`No results for: ${hero}` );
  });

  test('should update the path on search submit', () => { 
    
    const hero = 'Spider Man';

    const wrapper = mount(
      <MemoryRouter initialEntries={[`/search`]}>
        <SearchScreen />
      </MemoryRouter>
    );
    
    wrapper.find('input').simulate('change', {
      target: {
        name: 'searchText',
        value: hero,
      },
    });

    wrapper.find('form').prop('onSubmit')({preventDefault(){}});
 
    expect(mockNavigate).toHaveBeenCalledWith(`?q=${hero}`);
  });
});