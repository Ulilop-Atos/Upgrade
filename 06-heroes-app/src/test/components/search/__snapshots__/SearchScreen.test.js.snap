// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <SearchScreen /> should display properly 1`] = `
Array [
  <h1>
    Searches
  </h1>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          className="btn btn-outline-primary mt-1"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className="alert alert-info"
      >
         Search a hero 
      </div>
    </div>
  </div>,
]
`;

exports[`Tests for <SearchScreen /> should use the query value as input and display a hero 1`] = `
Array [
  <h1>
    Searches
  </h1>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Search a hero"
          type="text"
          value="Spider Man"
        />
        <button
          className="btn btn-outline-primary mt-1"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className="col animate__animated animate__fadeIn"
      >
        <div
          className="card"
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-4"
            >
              <img
                alt="Spider Man"
                className="card-img"
                src="/assets/marvel-spider.jpg"
              />
            </div>
            <div
              className="col-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Spider Man
                </h5>
                <p
                  className="card-text"
                >
                  Peter Parker
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Amazing Fantasy #15
                  </small>
                </p>
                <a
                  href="/hero/marvel-spider"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Tests for <SearchScreen /> should use the query value as input and display a not found message 1`] = `
Array [
  <h1>
    Searches
  </h1>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Search a hero"
          type="text"
          value="Crazy hero name"
        />
        <button
          className="btn btn-outline-primary mt-1"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className="alert alert-danger"
      >
         No results for: 
        Crazy hero name
         
      </div>
    </div>
  </div>,
]
`;
